internal/node/node_launcher.sh

```diff
@@ -129,4 +129,22 @@ for ARG in "${ALL_ARGS[@]}"; do
   esac
 done
 
-exec "${node}" "${NODE_OPTIONS[@]}" "${script}" "${ARGS[@]}"
+set +e
+"${node}" "${NODE_OPTIONS[@]}" "${script}" "${ARGS[@]}"
+RESULT="$?"
+set -e
+
+readonly EXPECTED_EXIT_CODE="TEMPLATED_expected_exit_code"
+if [ "${EXPECTED_EXIT_CODE}" -ne "0" ]; then
+  if (( ${RESULT} != ${EXPECTED_EXIT_CODE} )); then
+    echo "Expected exit code to be ${EXPECTED_EXIT_CODE}, but got ${RESULT}" >&2
+    # This exit code is handled specially by Bazel:
+    # https://github.com/bazelbuild/bazel/blob/486206012a664ecb20bdb196a681efc9a9825049/src/main/java/com/google/devtools/build/lib/util/ExitCode.java#L44
+    readonly BAZEL_EXIT_TESTS_FAILED = 3;
+    exit ${BAZEL_EXIT_TESTS_FAILED}
+  else
+    exit 0
```

Comment:
can just fall through in this case, don't need an else clause right?

Reply:
yes, since exit code (`$RESULT`) != 0, we need to `exit 0` since the correct non-zero exit code found

Reply:
right. I should use the program exit code if it is non-zero and not expected. updated this.